### ISSUE TITLE
Password Reset Functionality Disabled

### ISSUE THREAD

**Issue by troyfmay** [catalyst/moodle-auth_enrolkey#16]
If enrollment key authentication is set on a student's profile, they are unable to change their password. The pass reset feature "appears" to work for them and admins. But the system refuses to accept any password beyond the pass used during registration. System admins also cannot alter passwords for these students.

I have discovered there must be additional script written into the auth.php file. By adding a new command to the existing code, I was able to restore the functionality for my students who are enrolled with enrollment key authentication. Here is that script addendum:

```
function user_login($username, $password) {
        global $CFG, $DB;
        if ($user = $DB->get_record('user', array('username' => $username, 'mnethostid' => $CFG->mnet_localhost_id))) {
            return validate_internal_user_password($user, $password);
        }

        return false;
    }

    function user_update_password($user, $newpassword) {
        $user = get_complete_user_data('id', $user->id);
        return update_internal_user_password($user, $newpassword);
    }

    function can_signup() {
        return true;
    }

    function can_change_password() {
        return true;
    }

    function can_reset_password() {
        return true;
    }
```

I fixed ours, happily. I'm sure you guys know what to do with this to fix your's. Hope this help!

**Comment by brendanheywood** [comment]
awesome @troyfennell - can you please have a go at making a pull request in github. Please make 1 git branch per issue, make a pull request against that branch, and if you include an issue link ie literally '#16' somewhere in the git commit message then it will properly link the code to this issue

**Comment by troyfmay** [comment]
Will do, Brendan. I will try to have that fixed up in the next 24 hours.

**Comment by Peterburnett** [comment]
Thanks @troyfmay for the fix. I have implemented this function and it did fix the issue. This has been merged into master in PR #24. Closing as fixed!
